File: app/src/main/java/by/toggi/rxbsuir/di/BsuirServiceModule.java
package by.toggi.rxbsuir.di;

import javax.inject.Singleton;

import by.toggi.rxbsuir.rest.BsuirService;
import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import retrofit2.Retrofit;

@Module
@InstallIn(SingletonComponent.class)
public class BsuirServiceModule {

    @Provides
    @Singleton
    BsuirService provide(Retrofit retrofit) {
        return retrofit.create(BsuirService.class);
    }
}

File: api/src/main/kotlin/by/toggi/rxbsuir/api/internal/BsuirServiceModule.kt
package by.toggi.rxbsuir.api.internal

import by.toggi.rxbsuir.api.BsuirClient
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@Module
@InstallIn(SingletonComponent::class)
internal interface BsuirServiceModule {

    @Binds
    fun OkHttpClientBsuirClient.bindBsuirService(): BsuirClient
}
